engine/src/main/java/io/camunda/zeebe/streamprocessor/ProcessingScheduleServiceImpl.java

```diff
@@ -16,23 +17,47 @@
 public class ProcessingScheduleServiceImpl implements ProcessingScheduleService {
 
   private final ActorControl actorControl;
+  private final StreamProcessorContext streamProcessorContext;
 
-  public ProcessingScheduleServiceImpl(final ActorControl actorControl) {
-    this.actorControl = actorControl;
+  public ProcessingScheduleServiceImpl(final StreamProcessorContext streamProcessorContext) {
+    actorControl = streamProcessorContext.getActor();
+    this.streamProcessorContext = streamProcessorContext;
   }
 
   @Override
   public void runDelayed(final Duration delay, final Runnable followUpTask) {
     scheduleOnActor(() -> actorControl.runDelayed(delay, followUpTask));
   }
 
+  @Override
+  public void runDelayed(final Duration delay, final Task task) {
+    runDelayed(delay, toRunnable(task));
+  }
+
   @Override
   public <T> void runOnCompletion(
       final ActorFuture<T> precedingTask, final BiConsumer<T, Throwable> followUpTask) {
     scheduleOnActor(() -> actorControl.runOnCompletion(precedingTask, followUpTask));
   }
 
+  @Override
+  public void runAtFixedRate(final Duration delay, final Task task) {
+    /* TODO preliminary implementation; with the direct access
+     * this only works because this class is scheduled on the same actor as the
+     * stream processor.
+     */
+    runAtFixedRate(delay, toRunnable(task));
```

Comment:
:question: I'm a bit confused, which method is called here. Is this a loop?

Reply:
Ah I see there is a default implementation in the interface :sweat_smile:

Reply:
see comment above on quality level of this class